src/dataviews/dataviews-factory.js

```diff
@@ -96,6 +96,7 @@ module.exports = Model.extend({
   _generateAttrsForDataview: function (layerModel, attrs, whitelistedAttrs) {
     attrs = _.pick(attrs, whitelistedAttrs);
     attrs.source = attrs.source || { id: layerModel.id };
+    attrs.layerName = attrs.layerName || layerModel.get('layer_name');
```

Comment:
Imagine that the name of the layer is changed... `layerName` attribute won't see these changes...